### DISCUSSION TITLE
Is using AI to optimize code uploading the code to OpenAI?

### DISCUSSION THREAD

**Discussion by NathanRose-ISED** [plasma-umass/scalene#707] [Q&A] [answered]
I would like to know if Scalene's code optimization suggestions generate the recommendations locally or send the code to OpenAI. This is for security

**Comment by emeryberger** [comment, marked as answer]
It uses OpenAI's API, which is [bound by their privacy policy](https://openai.com/enterprise-privacy); they do not train on the data sent via this API and you retain ownership of that data.
